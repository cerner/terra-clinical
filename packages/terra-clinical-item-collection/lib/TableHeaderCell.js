'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('./ItemCollection.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * The column type of the header cell.
   **/
  columnType: _propTypes2.default.string
};

var defaultProps = {
  columnType: ''
};

var TableHeaderCell = function TableHeaderCell(_ref) {
  var columnType = _ref.columnType,
      customProps = _objectWithoutProperties(_ref, ['columnType']);

  var attributes = _extends({}, customProps);
  attributes.className = (0, _classnames2.default)(['terraClinical-ItemCollection-column', _defineProperty({}, 'terraClinical-ItemCollection-column--' + columnType, columnType)]);

  return _react2.default.createElement('th', attributes);
};

TableHeaderCell.propTypes = propTypes;
TableHeaderCell.defaultProps = defaultProps;

exports.default = TableHeaderCell;