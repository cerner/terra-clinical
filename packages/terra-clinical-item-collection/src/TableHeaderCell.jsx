import React, { PropTypes } from 'react';
import classNames from 'classnames';
import './ItemCollection.scss';

const propTypes = {
  /**
   * The column width of the header cell.
   **/
  columnWidth: PropTypes.string,
};

const defaultProps = {
  columnWidth: 'small',
};

const TableHeaderCell = ({ columnWidth, ...customProps }) => {
  const attributes = Object.assign({}, customProps);
  attributes.className = classNames(['terraClinical-ItemCollection-column-width',
    { [`terraClinical-ItemCollection-column-width-${columnWidth}`]: columnWidth },
  ]);

  return (
    <th {...attributes} />
  );
};

TableHeaderCell.propTypes = propTypes;
TableHeaderCell.defaultProps = defaultProps;

export default TableHeaderCell;
