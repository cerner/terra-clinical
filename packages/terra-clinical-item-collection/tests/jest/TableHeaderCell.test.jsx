import React from 'react';
import Table from 'terra-table';
import TableHeaderCell from '../../lib/_TableHeaderCell';

// Constants
const headerData1 = <TableHeaderCell columnType="accessory" key="accessory" id="accessory" />;
const headerData2 = <TableHeaderCell columnType="comment" key="comment" id="comment" />;
const headerData3 = <TableHeaderCell columnType="display" key="display" id="display" />;
const header = [headerData1, headerData2, headerData3];

// Snapshot test
it('should render accessory, comment and display table headers', () => {
  const defaultTableHeader = <Table.Header style={{ visibility: 'hidden' }}>{header}</Table.Header>;
  const tableHeader = render(defaultTableHeader);
  expect(tableHeader).toMatchSnapshot();
});
