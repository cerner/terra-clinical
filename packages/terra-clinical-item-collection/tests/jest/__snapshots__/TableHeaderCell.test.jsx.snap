exports[`test should render accessory, comment and display table headers 1`] = `
<thead
  style="visibility:hidden;">
  <tr>
    <th
      class="column-accessory"
      id="accessory" />
    <th
      data-terra-clinical-item-collection-column="comment"
      id="comment" />
    <th
      data-terra-clinical-item-collection-column="display"
      id="display" />
  </tr>
</thead>
`;
